apis/policies/v1alpha2/clusteradmissionpolicy_types.go

```diff
@@ -157,21 +157,39 @@ const (
 	// PolicyServerConfigMapReconciled represents the condition of the
 	// Policy Server ConfigMap reconciliation
 	PolicyServerConfigMapReconciled PolicyConditionType = "PolicyServerConfigMapReconciled"
-	// PolicyServerWebhookConfigurationReconciled represents the
-	// condition of the Policy Server WebhookConfiguration
-	// reconciliation
-	PolicyServerWebhookConfigurationReconciled PolicyConditionType = "PolicyServerWebhookConfigurationReconciled"
+	// ClusterAdmissionPolicyActive represents the condition of the Policy
+	// admission webhook being registered
+	ClusterAdmissionPolicyActive PolicyConditionType = "active"
+)
+
+// +kubebuilder:validation:Enum=unscheduled;unschedulable;pending;active
+type ClusterAdmissionPolicyStatusEnum string
+
+const (
+	// ClusterAdmissionPolicyStatusUnscheduled is a transient state that will continue
+	// to unschedulable or pending. This is the default state.
+	ClusterAdmissionPolicyStatusUnscheduled ClusterAdmissionPolicyStatusEnum = "unscheduled"
+	// ClusterAdmissionPolicyStatusUnschedulable informs that policy server where to
+	// schedule the policy is not available
+	ClusterAdmissionPolicyStatusUnschedulable ClusterAdmissionPolicyStatusEnum = "unschedulable"
+	// ClusterAdmissionPolicyStatusPending informs that the policy server exists,
+	// we are reconciling all resources
+	ClusterAdmissionPolicyStatusPending ClusterAdmissionPolicyStatusEnum = "pending"
+	// ClusterAdmissionPolicyStatusActive informs that the k8s API server should be
+	// forwarding admission review objects to the policy
+	ClusterAdmissionPolicyStatusActive ClusterAdmissionPolicyStatusEnum = "active"
 )
 
 // ClusterAdmissionPolicyStatus defines the observed state of ClusterAdmissionPolicy
 type ClusterAdmissionPolicyStatus struct {
-	// PolicyActive represents whether this AdmissionPolicy is active,
-	// such that the Kubernetes API server should be forwarding
-	// admission review objects to the policy
-	PolicyActive bool `json:"policyActive"`
+	// PolicyStatus represents whether this ClusterAdmissionPolicy is unscheduled,
+	// unschedulable, pending, or active.
+	// +kubebuilder:default:=unscheduled
+	PolicyStatus ClusterAdmissionPolicyStatusEnum `json:"policyStatus"`
 	// Conditions represent the observed conditions of the
 	// ClusterAdmissionPolicy resource.  Known .status.conditions.types
 	// are: "PolicyServerSecretReconciled",
+	// "PolicyServerBound",
```

Comment:
This is a leftover, there shouldn't be a `PolicyServerBound` and there should be a `ClusterAdmissionPolicyActive`.
This comment seems to be outdated, or I don't understand why policies can have conditions that are for the PolicyServer.

Reply:
Fixed in https://github.com/kubewarden/kubewarden-controller/pull/76/commits/eb9fe88174d48da76d048841f63a8c820149550d.

Reply:
I think we just set the ClusterAdmissionPolicyActive condition in the ClusterAdmissionPolicy, I am going to check it in my local env